appscale/tools/appscale_tools.py

```diff
@@ -505,14 +505,26 @@ def run_instances(cls, options):
       AppScaleLogger.log("\nPlease wait for AppScale to prepare your machines "
                          "for use. This can take few minutes.")
 
+      head_node.public_ip = public_ips[0]
+      head_node.private_ip = private_ips[0]
+      head_node.instance_id = instance_ids[0]
+
+      # Remove head node from node_layout nodes while updating nodes.
+      # TODO: Change the logic here to do two calls, one for head node(s)
+      # and another for scale set(s).
+      node_layout.nodes.remove(head_node)
+
       # Set newly obtained node layout info for this deployment.
-      for i, _ in enumerate(instance_ids):
-        node_layout.nodes[i].public_ip = public_ips[i]
-        node_layout.nodes[i].private_ip = private_ips[i]
-        node_layout.nodes[i].instance_id = instance_ids[i]
+      for i, _ in enumerate(instance_ids[1:]):
+        if "shadow" in node_layout.nodes[i].roles:
```

Comment:
Sorry, I still don't understand. If the head node is the shadow node, then it has already been removed from `node_layout.nodes`, so this would never get hit. If there are multiple shadow nodes, then wouldn't this prevent other nodes with that role from being assigned an IP address?